x/stake/keeper/validator.go

```diff
@@ -303,6 +303,8 @@ func (k Keeper) UpdateBondedValidators(ctx sdk.Context,
 			// TODO benchmark if we should read the current power and not write if it's the same
 			if bondedValidatorsCount == int(maxValidators) { // is cliff validator
 				k.setCliffValidator(ctx, validator, k.GetPool(ctx))
+			} else {
+				k.clearCliffValidator(ctx)
```

Comment:
Presumably we only need to clear the cliff validator if it was previously set (we had 100 validators) and it should be set no longer (we now have 99)?

Reply:
correct - I'll add this check in here

Reply:
addressed